api/libproject/devstart.rb

```diff
@@ -2408,24 +2424,26 @@ def deploy(cmd_name, args)
   common = Common.new
   common.status "Running database migrations..."
   ENV.update(read_db_vars(gcc))
-  # Note: `gcc` does not get correctly initialized with 'op.opts.account' so we need to be explicit
-  migrate_database(gcc, op.opts.account, op.opts.dry_run)
-  if (op.opts.key_file)
-    ENV["GOOGLE_APPLICATION_CREDENTIALS"] = op.opts.key_file
-  end
-  load_config(gcc.project, op.opts.dry_run)
-  cdr_config_file = must_get_env_value(gcc.project, :cdr_config_json)
-  update_cdr_config_for_project("config/#{cdr_config_file}", op.opts.dry_run)
+  ServiceAccountContext.new(gcc.project).run do
+    # Note: `gcc` does not get correctly initialized with 'op.opts.account' so we need to be explicit
+    migrate_database(gcc, op.opts.account, op.opts.dry_run)
+    if (op.opts.key_file)
+      ENV["GOOGLE_APPLICATION_CREDENTIALS"] = op.opts.key_file
+    end
```

Comment:
@dmohs these are the lines introduced in https://github.com/all-of-us/workbench/pull/7566

Reply:
This PR demonstrates a more complete understanding of the underlying problem than the PR that introduced this change, so my preference would be to remove it.

Testing is tricky. I don't feel confident I can replicate the Circle environment, so I generally just watch the builds and debug as needed. When you open up a completed job, there is usually an option to "Re-run with SSH enabled." I use this option extensively, making changes to files via Vim or otherwise to discover the root cause. When you merge this PR, your commit will have an icon indicator next to it that shows deploy status. Those are the jobs I would re-run with SSH enabled. Circle's output logs generally show the exact command executed, which you can use to reproduce.